### DISCUSSION TITLE
app.R fail on ShinyServer

### DISCUSSION THREAD

**Discussion by kyle-power** [ThinkR-open/golem#654] [Q&A] [answered]
Hey @ColinFay 
This feels like such a dumb question, but my `app.R` file is failing to start my `shiny` application. I'm using the open source linux version of `shiny-server` which is returning the following error during launch...
```{r}
> runApp()
Warning in loadSupport(appDir, renv = sharedEnv, globalrenv = NULL) :
  Loading R/ subdirectory for Shiny application, but this directory appears to contain an R package. 
  Sourcing files in R/ may cause unexpected behavior.
Error in Stack(horizontal = TRUE, horizontalAlign = "space-between", tokens = list(childrenGap = 20),  : 
  could not find function "Stack"
```
I don't know why `shiny-server` is calling `runApp()` instead of my instance of `run_app()`. I am using the standard `{golem}` shinyserver file...
```{r}
# Launch the ShinyApp (Do not remove this comment)
# To deploy, run: rsconnect::deployApp()
# Or use the blue button on top of this file

pkgload::load_all(export_all = FALSE,helpers = FALSE,attach_testthat = FALSE)
options( "golem.app.prod" = TRUE)
beast.dev::run_app() # add parameters here (if any)
```
The error makes me believe that the `package:namespace` is not loading with `pkgload::load_all()`... but I am not sure what is going on or how to even troubleshoot. I've spent hours Googling with no luck. 
This sounds related to https://github.com/ThinkR-open/golem/issues/33 but I couldn't figure it out. Any help is much appreciated!

**Comment by VincentGuyader** [comment, marked as answer]
HI, 

which golem version do you have ?
can you try to add a file `_disable_autoload.R` in your R folder and add options(shiny.autoload.r = FALSE) in your app.R file

see https://github.com/ThinkR-open/golem/issues/468 for context


Regards

**Reply by kyle-power** [reply]
I am using `{golem 0.2.1}` from CRAN, with `{shiny 1.6.0}`.
Adding the `_disable_autoload.R` with `options(shiny.autoload.r = FALSE)` fixed the issue!! 
Thank you so much for the quick response!!

**Reply by VincentGuyader** [reply]
you're welcome

**Reply by ColinFay** [reply]
Yes, that’s not a {golem} issue per-se but it’s due to a change in the recent versions of {shiny} :)

For future reference, you can call golem::disable_autoload() to automatically add the file — and there is no need for the option() to be set in there, the file itself is sufficient :)

Colin
